main_f4.c

```diff
@@ -377,7 +475,15 @@ main(void)
         }
 #endif
 
-#ifdef INTERFACE_USB
+	/* 
+	 * Check if the force-bootloader pins are strapped; if strapped,
+	 * don't try booting.
+	 */
+	if (board_test_force_pin()) {
```

Comment:
Previously, the bootloader would not have been activated on boot if the pins were strapped. It would only stay in the bootloader after a bootloader timeout.

I have now put this check here so if the pins are strapped it will trigger the bootlaoder. Is this the functionality that you want?